infrastructure/balsa_queue/balsa_queue.cc

```diff
@@ -7,23 +7,56 @@
 
 namespace jet {
 
+namespace {
+  double nanoseconds_to_seconds(double nanoseconds) {
+    return nanoseconds / 1000000000.0;
+  }
+}
+
 void BalsaQ::set_comms_factory(std::unique_ptr<CommsFactory> comms_factory) {
   comms_factory_ = std::move(comms_factory);
 }
 
 void BalsaQ::base_init() {
   bq_state_publisher_ = make_publisher("bq_state");
+  timestamp_last_report_ = get_current_time();
 }
 
 void BalsaQ::base_loop() {
   publish_state();
+  ++loop_count_since_last_report_;
 }
 
 void BalsaQ::publish_state() {
   static Timestamp last_state_publish_time_{0};
   if (last_state_publish_time_ + Duration::from_seconds(1) < get_current_time() || last_state_publish_time_ == 0) {
     BQStateMessage message;
     message.bq_instance_name = bq_name_;
+
+    if (loop_durations_.size() > 0) {
+      auto max_loop_time_it = std::max_element(loop_durations_.begin(), loop_durations_.end());
+      if (max_loop_time_it != loop_durations_.end()){
+        message.loop_execution_times.max_time = *max_loop_time_it;
+      }
+
+      auto min_loop_time_it = std::min_element(loop_durations_.begin(), loop_durations_.end());
+      if (min_loop_time_it != loop_durations_.end()){
+        message.loop_execution_times.min_time = *min_loop_time_it;
+      }
+
+      size_t median_element_index = loop_durations_.size() / 2;
```

Comment:
This can be const

[nit] The `2` should be `2u`

Reply:
:+1: